app/src/components/market/sections/market_list/market_home_container.tsx

```diff
@@ -106,32 +107,43 @@ const MarketHomeContainer: React.FC = () => {
   useEffect(() => {
     if (loading) {
       setMarkets(markets => (RemoteData.hasData(markets) ? RemoteData.reloading(markets.data) : RemoteData.loading()))
-    } else if (error) {
-      setMarkets(RemoteData.failure(error))
     } else if (fetchedMarkets) {
       const { fixedProductMarketMakers } = fetchedMarkets
 
       setMarkets(RemoteData.success(wrangleResponse(fixedProductMarketMakers, context.networkId)))
-      if (fixedProductMarketMakers.length === 0) {
+
+      if (fixedProductMarketMakers.length < PAGE_SIZE) {
         setMoreMarkets(false)
       }
+
+      setIsFiltering(false)
+    } else if (error) {
+      setMarkets(RemoteData.failure(error))
+      setIsFiltering(false)
     }
-  }, [fetchedMarkets, loading, error, context.networkId])
+  }, [fetchedMarkets, loading, error, context.networkId, markets])
 
   const onFilterChange = useCallback((filter: any) => {
-    setMoreMarkets(true)
     setFilter(filter)
+    setIsFiltering(true)
```

Comment:
I wonder if, instead of adding this flag, we could set markets to `RemoteData.loading()` here. Anyway, not a blocker, but it seems like `isFiltering` is kind of redundant.

Reply:
Well, I didn't know we could do that.

Tried to make the UI display a spinner every time there was an ongoing `RemoteData.loading()` but it seemed to trigger only when loading data / more data, not when filtering it.